### PR TITLE
kubelet/apis/config/validation: improve unit test coverage

### PR DESCRIPTION
**Pull request by tmilox**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

If applied this commit will improve the unit tests coverage of `pkg/kubelet/apis/config/validation`.

- before

Screenshot 2024-05-23 at 11 02 07

- after

Screenshot 2024-05-23 at 11 01 41

ps: Unfortunately I cannot bring it to 100% due to the `validateKubeletOSConfiguration` that returns an error only under Windows.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/apis/config/validation/validation_test.go [hunk 1]
```diff
@@ -598,6 +598,27 @@ func TestValidateKubeletConfiguration(t *testing.T) {
...
+			return conf
+		},
+		errMsg: "logging.format: Invalid value: \"invalid\": Unsupported log format",
+	}, {
+		name: "invalid FeatureGate",
+		configure: func(conf *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
+			conf.FeatureGates["invalid"] = true
+			return conf
+		},
+		errMsg: "unrecognized feature gate: invalid",
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/validation/validation_test.go:621]: /lgtm

**Comment by bart0sh** [comment]
/assign @mrunalp @derekwaynecarr @dchen1107 
for approval

**Comment by pacoxu** [comment]
> /assign @mrunalp @derekwaynecarr @dchen1107
> for approval
ACK
/lgtm

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Comment by pacoxu** [comment]
ping @kubernetes/api-approvers to approve
- sig node has approved this, and the PR is only about UT, not api change.

**Pull request merged**